sdk/digitaltwins/test-resources.bicep

```diff
@@ -9,23 +9,24 @@ param baseName string = resourceGroup().name
 @description('The location of the resource. By default, this is the same as the resource group.')
 param location string = resourceGroup().location
 
+@description('A new GUID used to identify the role assignment')
+param roleNameGuid string = newGuid()
+
 var adtOwnerRoleDefinitionId = '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/bcd981a7-7f74-457b-83e1-cceb9e632ffe'
 
-resource digitaltwin 'Microsoft.DigitalTwins/digitalTwinsInstances@2020-03-01-preview' = {
+resource digitaltwin 'Microsoft.DigitalTwins/digitalTwinsInstances@2020-12-01' = {
     name: baseName
     location: location
-    sku: {
-        name: 'S1'
-    }
     properties: {}
 }
 
-resource digitaltwinRoleAssignment 'Microsoft.DigitalTwins/digitalTwinsInstances/providers/roleAssignments@2020-03-01-preview' = {
-    name: '${digitaltwin.name}/Microsoft.Authorization/${guid(uniqueString(baseName))}'
+resource digitaltwinRoleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+    name: roleNameGuid
```

Comment:
Any reason why this doesn't match the format that was there previously?

Reply:
This has to do with name segments in the resource name, and its mapping to the actual resource created. Forward slashes `/` are used to delimit resource types / segments, so the previous format would imply creating a sub-resource that is invalid in this case (which the bicep interpreter also catches). We can still keep the `$guid(uniqueString(baseName))` for the resource name, though, if you prefer, instead of just a regular guid.